core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -102,7 +108,10 @@ protected DefaultDataContainer(int concurrencyLevel, int maxEntries, EvictionStr
          default:
             throw new IllegalArgumentException("No such eviction strategy " + strategy);
       }
-      entries = new BoundedConcurrentHashMap<Object, InternalCacheEntry>(maxEntries, concurrencyLevel, eviction, evictionListener);
+
+      entries = new BoundedConcurrentHashMapV8<Object, InternalCacheEntry>(
```

Comment:
The V8 version concurrencyLevel isn't needed as it tries to keep a load factor of .75 at all times.